tests/test_odp_zaius_graphql_api_manager.py

```diff
@@ -47,9 +48,75 @@ def test_fetch_qualified_segments__success(self):
 
         self.assertEqual(response, ['a', 'b'])
 
+    def test_fetch_qualified_segments__node_missing(self):
+        with mock.patch('requests.post') as mock_request_post, \
+                mock.patch('optimizely.logger') as mock_logger:
+            mock_request_post.return_value = \
+                self.fake_server_response(status_code=200, _content=self.node_missing_response_data.encode('utf-8'))
+
+            api = ZaiusGraphQLApiManager(logger=mock_logger)
+            api.fetch_segments(api_key=self.api_key,
+                               api_host=self.api_host,
+                               user_key=self.user_key,
+                               user_value=self.user_value,
+                               segments_to_check=['dummy1', 'dummy2', 'dummy3'])
+
+        mock_request_post.assert_called_once()
+        mock_logger.error.assert_called_once_with('Audience segments fetch failed (decode error).')
+
+    def test_fetch_qualified_segments__name_missing(self):
+        with mock.patch('requests.post') as mock_request_post, \
+                mock.patch('optimizely.logger') as mock_logger:
+            mock_request_post.return_value = \
+                self.fake_server_response(status_code=200, _content=self.name_missing_response_data.encode('utf-8'))
+
+            api = ZaiusGraphQLApiManager(logger=mock_logger)
+            api.fetch_segments(api_key=self.api_key,
+                               api_host=self.api_host,
+                               user_key=self.user_key,
+                               user_value=self.user_value,
+                               segments_to_check=['dummy1', 'dummy2', 'dummy3'])
+
+        mock_request_post.assert_called_once()
+        mock_logger.error.assert_called_once_with('Audience segments fetch failed (decode error).')
+
+    def test_fetch_qualified_segments__state_missing(self):
+        with mock.patch('requests.post') as mock_request_post, \
+                mock.patch('optimizely.logger') as mock_logger:
+            mock_request_post.return_value = \
+                self.fake_server_response(status_code=200, _content=self.state_missing_response_data.encode('utf-8'))
+
+            api = ZaiusGraphQLApiManager(logger=mock_logger)
+            api.fetch_segments(api_key=self.api_key,
+                               api_host=self.api_host,
+                               user_key=self.user_key,
+                               user_value=self.user_value,
```

Comment:
I think these 2 tests are covered by the following (mixed_missing_keys). Can be removed.